src/ComputeSharp.Graphics/Buffers/ConstantBuffer{T}.cs

```diff
@@ -80,36 +69,23 @@ public override void GetData(Span<T> span, int offset, int count)
         }
 
         /// <inheritdoc/>
-        public override void SetData(ReadOnlySpan<T> span, int offset, int count)
+        /// <inheritdoc/>
+        public override unsafe void SetData(ReadOnlySpan<T> span, int offset, int count)
         {
+            using MappedResource resource = MapResource();
             if (IsPaddingPresent)
             {
-                // Create the temporary array
-                byte[] temporaryArray = ArrayPool<byte>.Shared.Rent(count * PaddedElementSizeInBytes);
-                ReadOnlySpan<byte> temporarySpan = temporaryArray.AsSpan(0, count * PaddedElementSizeInBytes); // Array pool arrays can be longer
-                ref T tin = ref MemoryMarshal.GetReference(span);
-                ref byte tout = ref MemoryMarshal.GetReference(temporarySpan);
-
-                // Copy the input data to the temporary array and add the padding
-                for (int i = 0; i < count; i++)
+                // Copy element-by-element
+                ref var src = ref MemoryMarshal.GetReference(span);
+                byte* untypedDest = (byte*)resource.Pointer;
+                for (var i = offset; i < count; i++)
                 {
-                    ref byte rtarget = ref Unsafe.Add(ref tout, i * PaddedElementSizeInBytes);
-                    Unsafe.As<byte, T>(ref rtarget) = Unsafe.Add(ref tin, i);
+                    *(T*)&untypedDest[i * GetPaddedSize()] = Unsafe.Add(ref src, i);
```

Comment:
Same as above, you actually changed the behavior here. The items are always copied from the start of the given `ReadOnlySpan<T>`, the `offset` parameter is only used to specify what offset to use within the GPU buffer to start writing items. But again the items from the input span are always read from the start.